core/src/main/scala/kafka/log/Log.scala

```diff
@@ -339,6 +339,8 @@ class Log(@volatile var dir: File,
   }
 
   private def recoverSegment(segment: LogSegment, leaderEpochCache: Option[LeaderEpochCache] = None): Int = lock synchronized {
+    if (dir == null)
+      throw new KafkaStorageException(s"The log for partition $topicPartition is offline")
```

Comment:
It seems that log failure event won't be handled until the log loading completes. So, this is probably unnecessary.  If this can indeed happen, loadLogs() doesn't handle KafkaStorageException. So, such an exception will kill the broker, which is probably not we want.

Reply:
You are right. I will update the patch as suggested.